src/legacy/core_plugins/embeddable_api/public/test_samples/embeddables/filterable_container.tsx

```diff
@@ -26,10 +26,16 @@ export interface FilterableContainerInput extends ContainerInput {
   filters: Filter[];
 }
 
-export interface InheritedChildrenInput {
+/**
+ * interfaces are not allowed to specify a sub-set of the required types until
+ * https://github.com/microsoft/TypeScript/issues/15300 is fixed so we use a type
+ * here instead
+ */
+// eslint-disable-next-line @typescript-eslint/prefer-interface
+export type InheritedChildrenInput = {
```

Comment:
@stacey-gammon this was required by:
```
src/legacy/core_plugins/embeddable_api/public/test_samples/embeddables/filterable_container.tsx:35:3 - error TS2344: Type 'InheritedChildrenInput' does not satisfy the constraint 'Partial<EmbeddableInput>'.
  Index signature is missing in type 'InheritedChildrenInput'.

35   InheritedChildrenInput,
      ~~~~~~~~~~~~~~~~~~~~~~
```

The underlying issue is that the first generic param of `Container` is `TChildInput extends Partial<EmbeddableInput> = {}`, and `EmbeddableInput` extends `TIndexSignature` which extends:

```
interface TIndexSignature {
  [key: string]: unknown;
}
```

So, since `InheritedChildrenInput` doesn't include `[key: string]: unknown` we get an error, but we don't get that error if `InheritedChildrenInput` and other similar types are written as `type`s instead of `interface`s... Thoughts?

Reply:
I think that makes sense to me.